File: java/src/card/Deck.java
package card;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Deck {
    private List<Card> bill;
    {
        bill = Arrays.stream(Suit.values()).flatMap(s -> IntStream.rangeClosed(1,13).mapToObj(i -> new Card(s,i))).collect(Collectors.toList());
        Collections.shuffle(bill);
    }

    public Card draw(){
        return bill.remove(0);
    }
}

File: scala/src/main/scala/Deck.scala
import scala.collection.mutable._
import scala.util.Random

case class Deck() {
  val bill = {
    val cards: Queue[Card] = Queue()
    for (s <- Suit.values; i <- 1 to 13) cards += Card(s, i)
    Random.shuffle(cards)
  }

  def draw = bill.dequeue
}
